lib/AST/ASTContext.cpp

```diff
@@ -2136,6 +2139,24 @@ ASTContext::getSelfConformance(ProtocolDecl *protocol) {
   return entry;
 }
 
+/// Produce the builtin conformance for some non-nominal to some protocol.
+BuiltinProtocolConformance *
+ASTContext::getBuiltinConformance(Type type, ProtocolDecl *protocol,
+                                ArrayRef<ProtocolConformanceRef> conformances) {
+  auto key = std::make_pair(type, protocol);
+  auto &builtinConformances =
+    getImpl().getArena(AllocationArena::Permanent).BuiltinConformances;
```

Comment:
I'm pretty sure we can end up with builtin conformances that involve type variables, and therefore this would need to be stored in a temporary arena. We should do the normal computation to figure out what area to use.